pallets/author-inherent/src/lib.rs

```diff
@@ -179,6 +186,14 @@ impl<T: Config> ProvideInherent for Module<T> {
 	type Error = InherentError;
 	const INHERENT_IDENTIFIER: InherentIdentifier = INHERENT_IDENTIFIER;
 
+	fn is_inherent_required(_: &InherentData) -> Result<Option<Self::Error>, Self::Error> {
+		// Return Ok(Some(_)) unconditionally because this inherent is required in every block
+		// If it is not found, throw an AuthorInherentRequired error.
+		Ok(Some(InherentError::Other(
+			sp_runtime::RuntimeString::Borrowed("AuthorInherentRequired"),
+		)))
+	}
+
```

Comment:
@4meta5 This is why I was wondering whether we still need the check in `on_finalize`.

See https://github.com/paritytech/substrate/pull/7941/ for my best understanding of how this trait works.

Reply:
I see. This should work, but still need to clear the storage item in `on_finalize` so it can be set in the next block. So can remove the `assert!(<Author>::take)` and replace it with `<Author>::kill()`

Reply:
@4meta5 Could you please investigate whether this cleanup is valid in another PR?